Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -633,4 +636,59 @@ Optional<Color> Bitmap::solid_color(u8 alpha_threshold) const
     return color;
 }
 
+static float color_distance_squared(Gfx::Color const& lhs, Gfx::Color const& rhs)
+{
+    int a = rhs.red() - lhs.red();
+    int b = rhs.green() - lhs.green();
+    int c = rhs.blue() - lhs.blue();
+    int d = rhs.alpha() - lhs.alpha();
+    return (a * a + b * b + c * c + d * d) / (4.0f * 255.0f * 255.0f);
+}
```

Comment:
I feel like this is a thing about `Color`, not `Bitmap`. It could live in `Color.h`/`Color.cpp`, like `Color::mixed_with()` or `Color::multiply()`